components/input/gpio/gpio_test.go

```diff
@@ -526,46 +515,16 @@ func TestGPIOInput(t *testing.T) {
 				target = 0
 				s.b.Analogs["analog1"].Set(0)
 			}
-			testutils.WaitForAssertion(t, func(tb testing.TB) {
+			testutils.WaitForAssertionWithSleep(t, 5*time.Millisecond, 25, func(tb testing.TB) {
 				tb.Helper()
 				state, err := s.dev.Events(s.ctx, map[string]interface{}{})
 				test.That(tb, err, test.ShouldBeNil)
 				test.That(tb, state["AbsoluteX"].Value, test.ShouldAlmostEqual, target, 0.005)
-				test.That(tb, atomic.LoadInt64(&s.axis1Callbacks), test.ShouldEqual, i+1)
-			})
-			s.axisMu.RLock()
-			test.That(t, s.axis1Time.Sub(startTime), test.ShouldBeBetween, 0*time.Millisecond, 110*time.Millisecond)
-			s.axisMu.RUnlock()
-		}
-
-		s.b.Analogs["analog2"].Set(1023)
```

Comment:
this whole test was removed

Reply:
Why was this whole test removed? Wasn't it doing something useful?

Reply:
i tested default vs. overriden poll rate. potential flakiness introduced by testing this way wasn't worth it.